include/boost/gil/image_processing/filter.hpp

```diff
@@ -60,6 +60,53 @@ void blur(
     box_filter(src_view, dst_view, kernel_size, anchor, true, option);
 }
 
+
+template <typename SrcView, typename DstView>
+void median_filter(
+    SrcView const& src_view,
+    DstView const& dst_view,
+    std::size_t kernel_size
+)
+{
+    static_assert(color_spaces_are_compatible
+    <
+        typename color_space_type<SrcView>::type,
+        typename color_space_type<DstView>::type
+    >::value, "Source and destination views must have pixels with the same color space");
+
+    std::size_t half_kernel_size = kernel_size / 2;
+    auto extended_img = extend_boundary(src_view, kernel_size / 2, boundary_option::extend_constant);
+    auto extended_view = subimage_view(
+            view(extended_img),
+            half_kernel_size,
+            half_kernel_size,
+            src_view.width(),
+            src_view.height()
+        );
+
+    //deciding output channel type and creating functor
+    using result_channel_t = typename channel_type<DstView>::type;
+
+    for (std::ptrdiff_t y = 0; y < extended_view.height(); y++)
+    {
+        typename DstView::x_iterator dst_it = dst_view.row_begin(y);
+
+        for (std::ptrdiff_t x = 0; x < extended_view.width(); x++)
+        {
+            static_transform(
+                subimage_view(extended_view, x - half_kernel_size, y - half_kernel_size, kernel_size, kernel_size),
+                dst_it[x],
```

Comment:
@miralshah365 First, `static_transform` accepts and operates on color base - [a container of color elements](https://www.boost.org/doc/libs/develop/libs/gil/doc/html/design/color_base.html) e.g. pixels, but not on image views.

Second, above you are mixing both, view as source with pixel as destination.

Since `layout_t` is a trait of color base (e.g. pixel) and not image view, then you are getting this particular compilation error:

```
gil/color_base_algorithm.hpp:106:78: error: no type named ‘layout_t’ in ‘class boost::gil::image_view<....>
```

Does it make sense and help?

Reply:
Yes, this was the problem. Correcting that solved all the issues with my code

Reply:
All the CI-s are happy now. Good job! 

I will review it over the weekend.